libs/dal/src/types/helpers.ts

```diff
@@ -1,5 +1,42 @@
 import { Types } from 'mongoose';
 
-export type ChangePropsValueType<T, K extends keyof T, V = Types.ObjectId> = Omit<T, K> & {
-  [P in K]: V;
-};
+/**
+ * The type of an ObjectKey identifier.
+ * This is the type of the `_id` field in a Mongoose document
+ *
+ * This is intentionally type widened to `string` to cater for
+ * cases where the `_id` field is a string instead of a `Types.ObjectId`
+ * TODO: Investigate and decide on approach to choose either `Types.ObjectId` or `string`
+ *
+ * @see https://mongoosejs.com/docs/schematypes.html#objectids
+ */
+export type ObjectIdType = Types.ObjectId | string;
+
+/**
+ * The type union of the PK identifier and the templater literal type of an FK identifier
+ */
+export type ObjectIdKey = '_id' | `_${string}Id`;
+
+/**
+ * The templater literal type of a Date key
+ */
+export type DateKey = `${string}At`;
+
+/**
+ * Flatten type T to make it easier to read
+ */
+export type Identity<T> = { [P in keyof T]: T[P] };
+
+/**
+ * Transform the values of T with keys in union of type U to type V
+ */
+export type TransformValues<T, U, V> = Identity<
+  Omit<T, Extract<keyof T, U>> & {
+    [P in keyof Pick<T, Extract<keyof T, U>>]: V;
+  }
+>;
+
+/**
+ * Transform the values of T with keys in union of type `ObjectIdKey` to type `Types.ObjectId`
+ */
+export type TransformEntityToDbModel<T> = TransformValues<TransformValues<T, ObjectIdKey, ObjectIdType>, DateKey, Date>;
```

Comment:
This was the old `ChangePropsValueTypes` generic, updated to use the new base generic, with a new Date property transform, and renamed to better fit the purpose.

Reply:
Love this one, now we won't need to do manual mapping for every entity.